pkg/cableengine/ipsec/strongswan.go

```diff
@@ -291,15 +312,52 @@ func (i *strongSwan) loadSharedKey(endpoint types.SubmarinerEndpoint, client *go
 	return nil
 }
 
-func runCharon(debug bool, logFile string) error {
+const charonConfTemplate = `
+	charon {
+		port = {{.ipSecIKEPort}}
+		port_nat_t = {{.ipSecNATTPort}}
+		make_before_break = yes
+		ignore_acquire_ts = yes
+	}
+`
+
+func (i *strongSwan) writeCharonConfig(path string) {
+	err := os.Remove(path)
+	if err != nil {
+		klog.Infof("Error deleting %s", path)
+	}
+
+	f, err := os.Create(path)
+	if err != nil {
+		klog.Fatalf("Error creating %s: %s", path, err)
+	}
+	i.renderCharonConfigTemplate(f)
```

Comment:
You need to close the FD after usage and check for errors on closure.

Reply:
Oh my, that's a good catch, Thanks,